lib/src/components/NxTag/__tests__/NxTag.test.tsx

```diff
@@ -4,124 +4,125 @@
-import 'jest-enzyme';
-import { shallow, mount } from 'enzyme';
-import { faPlusCircle, faTimesCircle } from '@fortawesome/free-solid-svg-icons';
+import React, { RefAttributes } from 'react';
 
-import NxTag, { NxSelectableTag } from '../NxTag';
-import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import NxTag, { NxSelectableTag, PublicProps, SelectableProps } from '../NxTag';
+
+import { rtlRender, rtlRenderElement, userEvent } from '../../../__testutils__/rtlUtils';
 
 describe('NxTag', function() {
-  const getShallowComponent = enzymeUtils.getShallowComponent(NxTag, { children: 'basic tag' });
+  type PropsWithRef = PublicProps & RefAttributes<HTMLLabelElement>;
+  const renderEl = rtlRenderElement<PropsWithRef>(NxTag, { children: 'basic tag '});
 
   it('renders NxTag with the `nx-tag` class', function() {
-    expect(getShallowComponent().children().find('.nx-tag')).toExist();
+    const el = renderEl()!;
+    expect(el.tagName).toBe('LABEL');
+    expect(el).toHaveClass('nx-tag');
   });
 
   it('forwards a ref', function() {
-    const ref = React.createRef<HTMLLabelElement>();
-    const component = mount(<><NxTag ref={ref}>Hello</NxTag></>);
-    expect(component.find('.nx-tag').getDOMNode()).toBe(ref.current);
+    const ref = React.createRef<HTMLLabelElement>(),
+        renderedEl = renderEl({ ref });
+
+    expect(ref.current).toBe(renderedEl);
   });
 
   it('correctly assigns supplied class', function() {
-    const classComponent = getShallowComponent({ className: 'foo' });
-    expect(classComponent.find('.nx-tag')).toHaveClassName('foo');
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
   it('correctly assigns supplied id', function() {
-    const idComponent = getShallowComponent({ id: 'test-id' });
-    expect(idComponent.find('.nx-tag')).toHaveProp('id', 'test-id');
+    const el = renderEl({ id: 'test-id' });
+    expect(el).toHaveAttribute('id', 'test-id');
   });
 
   it('renders the supplied text inside .nx-tag__text', function() {
-    const tagChildren = getShallowComponent({ children: 'tag text' });
-    expect(tagChildren.find('.nx-tag__text')).toHaveText('tag text');
-
-    expect(tagChildren.children().first()).toMatchSelector('label');
-    // expect(tagChildren.children().at(1)).toMatchSelector('span');
+    const el = renderEl({ children: 'tag text' });
+    expect(el).toHaveTextContent('tag text');
   });
 
   it('sets the nx-selectable-color--indigo class if no color prop is passed', function() {
-    expect(getShallowComponent().find('.nx-selectable-color--indigo')).toExist();
+    const el = renderEl();
+    expect(el).toHaveClass('nx-selectable-color--indigo');
   });
 
   it('sets the color class using the color if it is provided', function() {
```

Comment:
Should be tested via visual tests; the classnames are impl details

Reply:
> Should be tested via visual tests; the classnames are impl details

Changed here: 4e903318fd82ddafd673824edd89ea5bee4d6d89